packages/cli-main/src/cli/services/upgrade.ts

```diff
@@ -0,0 +1,174 @@
+import {error, file, os, output, path, system} from '@shopify/cli-kit'
+import {
+  addNPMDependencies,
+  findUpAndReadPackageJson,
+  checkForNewVersion,
+  DependencyType,
+  getPackageManager,
+} from '@shopify/cli-kit/node/node-package-manager'
+
+// Canonical list of oclif plugins that should be installed globally
+const globalPlugins = ['@shopify/theme']
+
+interface PackageJsonContents {
```

Comment:
What about reusing [this model](https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/node/node-package-manager.ts#L210) from `@shopify/cli-kit`?